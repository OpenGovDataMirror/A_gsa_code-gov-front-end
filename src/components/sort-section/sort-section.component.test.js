import React from 'react'
import { shallow } from 'enzyme'

import SortSection from 'components/sort-section/sort-section.component'

const props = {
  onSortChange: jest.fn(),
  options: [
    { value: 'opt-1', label: 'label-1', selected: false },
    { value: 'opt-2', label: 'label-2', selected: true },
  ]
}

let wrapper
let instance
describe('components - SortSection', () => {
  beforeEach(() => {
    wrapper = shallow(<SortSection {...props} />)
    instance = wrapper.instance()
  })

  describe('onSortChange', () => {
    it('should call `onSortChange` with the value', () => {
      instance.onSortChange({ target: { value: 'test-value' } })
      expect(props.onSortChange).toBeCalledWith('test-value')
    })

    it('should not throw if `onSortChange` not provided', () => {
      wrapper.setProps({ onSortChange: undefined })
      expect(() => instance.onSortChange({ target: { value: 'test-value' } })).not.toThrow()
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render all the options', () => {
      expect(wrapper.find('option').length).toBe(props.options.length)
    })
  })
})