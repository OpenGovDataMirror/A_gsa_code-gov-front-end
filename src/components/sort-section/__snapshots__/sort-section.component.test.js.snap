// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - SortSection render should render correctly 1`] = `
<div
  className="grid-row"
>
  <h2
    aria-hidden="true"
    className="margin-top-4"
  >
    Sort by
  </h2>
  <form
    className="usa-form"
  >
    <label
      className="usa-label usa-sr-only"
      htmlFor="sort-options"
    >
      Sort by
    </label>
    <select
      className="usa-select margin-left-1 radius-md"
      id="sort-options"
      onChange={[Function]}
      value="opt-2"
    >
      <option
        key="opt-1"
        value="opt-1"
      >
        label-1
      </option>
      <option
        key="opt-2"
        value="opt-2"
      >
        label-2
      </option>
    </select>
  </form>
</div>
`;
